.github/workflows/ci.yml

```diff
@@ -30,313 +30,74 @@ permissions:
   issues: write
 
 jobs:
-  get-runner-labels:
-    name: Get Runner Labels
-    uses: ./.github/workflows/get-runner-labels.yml
 
   check-changed:
     runs-on: ubuntu-latest
     name: Check Source Changed
     outputs:
-      changed: ${{ steps.filter.outputs.changed == 'true' }}
-      document_changed: ${{ steps.document_filter.outputs.changed == 'true' }}
+      code_changed: ${{ steps.filter.outputs.code_changed == 'true' }}
+      document_changed: ${{ steps.filter.outputs.document_changed == 'true' }}
     steps:
       - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
       - uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 # v3
         id: filter
         with:
           predicate-quantifier: "every"
           filters: |
-            changed:
+            code_changed:
               - "!**/*.md"
               - "!**/*.mdx"
               - "!website/**"
-      - uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 # v3
-        id: document_filter
-        with:
-          predicate-quantifier: "every"
-          filters: |
-            changed:
+            document_changed:
               - "website/**"
 
   test-linux:
     name: Test Linux
-    needs: [get-runner-labels, check-changed]
-    if: ${{ needs.check-changed.outputs.changed == 'true' }}
+    needs: [ check-changed ]
+    if: ${{ needs.check-changed.outputs.code_changed }}
     uses: ./.github/workflows/reusable-build.yml
     with:
       target: x86_64-unknown-linux-gnu
-      runner: ${{ needs.get-runner-labels.outputs.LINUX_RUNNER_LABELS }}
+      runner: ${{ vars.LINUX_SELF_HOSTED_RUNNER_LABELS }}
```

Comment:
vars.LINUX_SELF_HOSTED_RUNNER_LABELS will not exist if you fork rspack and provide a PR
```
runner: ${{ vars.LINUX_SELF_HOSTED_RUNNER_LABELS || '"ubuntu-22.04"' }}
```